### PR TITLE
Knative autoscaling

### LINKED ISSUE #10641: Kubernetes resources get duplicated then trying to use existing resources in src/main/kubernetes
**Describe the bug**
I'm trying to customise the generated Kubernetes resources of the Kubernetes extension by providing base resources. 

From the documentation I understand that if the custom resource has the same name like the generated one it will be merged together into one resource. 
However, I find the resource duplicated instead.

Not sure if something is wrong or if I have an error in reasoning.
Can someone shed some light please?

**To Reproduce**

1. Create new project from Archetype "kubernetes-quickstart" (like documented here: https://quarkus.io/guides/deploying-to-kubernetes)
2. Add a kubernetes.yaml to "src/main/kubernetes" with the following content: 
```
apiVersion: v1
kind: Namespace
metadata:
  name: quickstart
---
apiVersion: v1
kind: Service
metadata:
  name: kubernetes-quickstart
  namespace: quickstart
```
3. Run Maven build: mvn package

**Expected behavior**
Generated Kubernetes resources contains one Service kind that contains the amended namespace 
```
---
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2020-07-10 - 11:17:13 +0000
  labels:
    app.kubernetes.io/name: kubernetes-quickstart
    app.kubernetes.io/version: 1.0-SNAPSHOT
  name: kubernetes-quickstart
  namespace: quickstart
spec:
  ports:
  - name: http
    port: 8080
    targetPort: 8080
  selector:
    app.kubernetes.io/name: kubernetes-quickstart
    app.kubernetes.io/version: 1.0-SNAPSHOT
  type: ClusterIP
---
``` 

**Actual behavior**
Generated Kubernetes resources contains two Service kinds with the same name (existing one + generated one)

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2020-07-10 - 11:17:13 +0000
  name: kubernetes-quickstart
  namespace: quickstart
---
apiVersion: v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2020-07-10 - 11:17:13 +0000
  labels:
    app.kubernetes.io/name: kubernetes-quickstart
    app.kubernetes.io/version: 1.0-SNAPSHOT
  name: kubernetes-quickstart
spec:
  ports:
  - name: http
    port: 8080
    targetPort: 8080
  selector:
    app.kubernetes.io/name: kubernetes-quickstart
    app.kubernetes.io/version: 1.0-SNAPSHOT
  type: ClusterIP
---
```

**Environment:**
Quarkus Version: 1.6.0.Final
Maven Verson: 3.6.3

### PR DESCRIPTION
**Pull request by iocanel**
This pull request adds autoscaling configuration for knative.
Also indirectly resolves: #10641

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AutoScalingConfig.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import io.quarkus.runtime.annotations.ConfigGroup;
+
+@ConfigGroup
+public class AutoScalingConfig {
+
+    /**
+     * The Autoscaler class.
+     * Knative Serving comes with its own autoscaler, the KPA (Knative Pod Autoscaler) but can also be configured to use
+     * Kubernetes’ HPA (Horizontal Pod Autoscaler) or even a custom third-party autoscaler.
+     * Possible values (kpa, hpa, default: kpa).
```
Comment (geoand) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AutoScalingConfig.java:15]: When you say default, you mean the default that knative itself uses if nothing is configured?

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AutoScalingConfig.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+     * Possible values (concurency, rps, cpu).
+     * 
+     * @return The cpu metric or NONE if no metric has been selected.
+     */
+    Optional<AutoScalingMetric> metric;
+
+    /**
+     * The autoscaling target.
+     * 
+     * @reutrn the selected target or zero if no target is selected.
```
Comment (geoand) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AutoScalingConfig.java:33]:
```suggestion
     * @return the selected target or zero if no target is selected.
```

**Comment by geoand** [comment]
I added some tiny comments, but more importantly:

Can we separate the commit that bumps dekorate (which I assume is the one that fixes #10641)? I would like to have that so we can then backport the commit into 1.6.x without having to bring in the new knative features

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AutoScalingConfig.java (hunk 1 above)
Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AutoScalingConfig.java:15]: Yes

**Comment by geoand** [comment]
Can we also add the issue number that the dekorate bump fixes to the commit message?

**Review by geoand — APPROVED** [review]
Excellent!

**Comment by geoand** [comment]
@gsmet from this PR, I would like to have the first commit (version bump) backported if possible.

Thanks

**Comment by gsmet** [comment]
I backported the version bump.

**Comment by geoand** [comment]
Thanks!

**Pull request merged**